core/src/rpc/aggregator.rs

```diff
@@ -386,106 +384,174 @@ impl ClementineAggregator for Aggregator {
+                        .await?
+                        .into_inner();
 
-                    for param in params {
-                        tx.send(param)
-                            .await
-                            .map_err(|_| output_stream_ended_prematurely())?;
+                    let mut params = Vec::new();
+                    while let Some(param) = stream.message().await? {
+                        params.push(param);
                     }
-
-                    client
-                        .set_operator(tokio_stream::wrappers::ReceiverStream::new(rx))
-                        .await?;
+                    tx.send(Some(params)).map_err(|e| {
+                        BridgeError::Error(format!("failed to forward operator params: {e}"))
+                    })?;
+                    Ok::<_, Status>(())
                 }
-
-                Ok::<_, tonic::Status>(())
-            }
-        }))
-        .await?;
-
-        tracing::info!("Collecting Winternitz public keys from watchtowers...");
-        let watchtower_params = try_join_all(self.watchtower_clients.iter().map(|client| {
-            let mut client = client.clone();
-
-            async move {
-                let mut responses = Vec::new();
-
-                let mut params_stream = client
-                    .get_params(Request::new(Empty {}))
-                    .await?
-                    .into_inner();
-                while let Some(response) = params_stream.message().await? {
-                    responses.push(response);
+            }))
+            .await?;
+            tx.send(None).map_err(|e| {
+                BridgeError::Error(format!("failed to close operator params stream: {e}"))
+            })?;
+            Ok::<_, Status>(())
+        });
+
+        let tx = operator_params_tx.clone();
+        let clients = self.verifier_clients.clone();
+        let set_operator_params_handle = tokio::spawn(async move {
+            tracing::info!("Informing verifiers for existing operators...");
+            try_join_all(clients.iter().map(|client| {
+                let mut client = client.clone();
+                let mut operator_params_rx = tx.subscribe();
+                async move {
+                    loop {
+                        match operator_params_rx.try_recv() {
+                            Ok(Some(params)) => {
+                                let (tx, rx) = channel(CHANNEL_CAPACITY);
+                                let stream = client.set_operator(ReceiverStream::new(rx));
+                                for param in params {
+                                    tx.send(param).await.map_err(|e| {
+                                        BridgeError::RPCStreamEndedUnexpectedly(format!(
+                                            "failed to propagate operator params: {e}"
+                                        ))
+                                    })?;
+                                }
```

Comment:
```suggestion
                                use futures::stream;
                                client.set_operator(stream::iter(params)).await?;
```
No need to create broadcast channels here, use Stream constructors. (also for the other calls where we're simply passing Vecs)

Reply:
Good one, thank you.

Reply:
(Just a follow up: `.await?` on `set_operator` seems to deadlock as well)

Reply:
weird, what's the difference from what you're doing? you're still sending all of the params and waiting at the end.

Reply:
nevermind, the `stream::iter` works fine